frontend/src/metabase/auth/components/ResetPasswordForm/ResetPasswordForm.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useMemo } from "react";
+import { t } from "ttag";
+import * as Yup from "yup";
+import _ from "underscore";
+import MetabaseSettings from "metabase/lib/settings";
+import Form from "metabase/core/components/Form";
+import FormProvider from "metabase/core/components/FormProvider";
+import FormInput from "metabase/core/components/FormInput";
+import FormSubmitButton from "metabase/core/components/FormSubmitButton";
+import FormErrorMessage from "metabase/core/components/FormErrorMessage";
+import { ResetPasswordData } from "metabase/auth/types";
+import {
+  PasswordFormMessage,
+  PasswordFormTitle,
+} from "./ResetPasswordForm.styled";
+
+const ResetPasswordSchema = Yup.object({
+  password: Yup.string()
+    .required(t`required`)
+    .test(async (value = "", context) => {
+      const error = await context.options.context?.onValidatePassword(value);
+      return error ? context.createError({ message: error }) : true;
+    }),
+  password_confirm: Yup.string()
+    .required(t`required`)
+    .oneOf([Yup.ref("password")], t`passwords do not match`),
+});
+
+interface ResetPasswordFormProps {
+  onValidatePassword: (password: string) => Promise<string | undefined>;
+  onSubmit: (data: ResetPasswordData) => void;
+}
+
+const ResetPasswordForm = ({
+  onValidatePassword,
+  onSubmit,
+}: ResetPasswordFormProps): JSX.Element => {
+  const initialValues = useMemo(
+    () => ({ password: "", password_confirm: "" }),
+    [],
+  );
```

Comment:
I don't think we need useMemo here? and I think it's less performant for static values like this.

Reply:
We don't need it here. However we *might* need it case we use `enableReinitialize` in formik. I just thought it's a good idea to make sure that `initialValues`'s reference is always stable.